dvc/utils/serialize/_py.py

```diff
@@ -0,0 +1,89 @@
+import ast
+
+from funcy import reraise
+
+from ._common import ParseError, _load_data
+
+
+class PYFileCorruptedError(ParseError):
+    def __init__(self, path, message="PY file structure is corrupted"):
+        super().__init__(path, message)
+
+
+def load_py(path, tree=None):
+    return _load_data(path, parser=parse_py, tree=tree)
+
+
+def parse_py(text, path):
+    """Parses text from .py file into Python structure."""
+    with reraise(SyntaxError, PYFileCorruptedError(path)):
+        tree = ast.parse(text, filename=path)
+
+    result = ast_tree_to_dict(tree)
+    return result
+
+
+def ast_tree_to_dict(tree):
+    """Parses ast trees to dict."""
+    result = {}
+    for _body in tree.body:
+        try:
+            if isinstance(_body, ast.Assign):
+                result.update(ast_assign_to_dict(_body))
+            elif isinstance(_body, ast.ClassDef):
+                result.update({_body.name: ast_tree_to_dict(_body)})
+            elif (
+                isinstance(_body, ast.FunctionDef) and _body.name == "__init__"
+            ):
+                result.update(ast_tree_to_dict(_body))
+        except ValueError:
+            continue
+        except AttributeError:
+            continue
+    return result
+
+
+def ast_assign_to_dict(assign):
+    result = {}
+
+    if len(assign.targets) == 1:
```

Comment:
Looks like this does not work with annotations based assignment.
```python
is_dir: bool = True
```

But, for now, we could totally move without it.

Reply:
I fixed it.